### ISSUE TITLE
[nightly] agent_stress_test failed

### ISSUE THREAD

**Issue by iycheng** [ray-project/ray#32873]
https://console.anyscale-staging.com/o/anyscale-internal/jobs/prodjob_jn8nx9xbw2adkp7653gcpg4xvj
https://console.anyscale-staging.com/o/anyscale-internal/jobs/prodjob_1nmr8u6czjxp1zjvr3n67l1ti8

```
2023-02-23 15:09:36,706 INFO worker.py:1243 -- Using address 10.0.15.187:6379 set in the environment variable RAY_ADDRESS
Traceback (most recent call last):
  File "workload.py", line 5, in <module>
    ray.init("auto")
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/_private/worker.py", line 1338, in init
    job_config.runtime_env
  File "/home/ray/anaconda3/lib/python3.7/site-packages/anyscale/snapshot_util.py", line 591, in env_hook
    os.environ["ANYSCALE_EXPERIMENTAL_INITIAL_JOB_ID"], runtime_env
  File "/home/ray/anaconda3/lib/python3.7/site-packages/anyscale/snapshot_util.py", line 513, in checkpoint_job
    urllib.request.urlretrieve(runtime_env["working_dir"], working_dir)
  File "/home/ray/anaconda3/lib/python3.7/urllib/request.py", line 247, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/home/ray/anaconda3/lib/python3.7/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/home/ray/anaconda3/lib/python3.7/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/home/ray/anaconda3/lib/python3.7/urllib/request.py", line 548, in _open
    'unknown_open', req)
  File "/home/ray/anaconda3/lib/python3.7/urllib/request.py", line 503, in _call_chain
    result = func(*args)
  File "/home/ray/anaconda3/lib/python3.7/urllib/request.py", line 1389, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: gcs>

Traceback (most recent call last):
  File "mem_check.py", line 64, in <module>
    assert False, "Job has failed."
AssertionError: Job has failed.
Subprocess return code: 1
[INFO 2023-02-23 15:09:51,915] anyscale_job_wrapper.py: 174  Process 1777 exited with return code 1.
[INFO 2023-02-23 15:09:51,916] anyscale_job_wrapper.py: 274  Finished with return code 1. Time taken: 18.725816346000016
[ERROR 2023-02-23 15:09:51,916] anyscale_job_wrapper.py: 62  Couldn't upload to s3: '/tmp/release_test_out.json' does not exist.
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
pytorch-lightning 1.5.10 requires setuptools==59.5.0, but you have setuptools 67.4.0 which is incompatible.
boto3 1.26.78 requires botocore<1.30.0,>=1.29.78, but you have botocore 1.27.6 which is incompatible.
azure-cli-core 2.29.1 requires requests[socks]~=2.25.1, but you have requests 2.28.2 which is incompatible.
autogluon-core 0.2.0 requires numpy==1.19.5, but you have numpy 1.21.6 which is incompatible.
aiobotocore 1.2.2 requires botocore<1.19.53,>=1.19.52, but you have botocore 1.27.6 which is incompatible.

```

**Comment by rickyyx** [comment]
Seems to be job runtime env related? cc @architkulkarni ?

**Comment by architkulkarni** [comment]
This looks like an unforeseen interaction with Workspaces where the Anyscale Job's runtime environment `working_dir` is being overwritten by something that doesn't make sense.  @shomilj do you know who could help here? Workspaces code shouldn't be being triggered by OSS release tests, right?

**Comment by architkulkarni** [comment]
`jobs_basic_working_dir` and `jobs_specify_num_gpus` are also failing with this error.  

What all these tests seem to have in common is that they launch a Ray Job from inside the entrypoint script with a local working_dir.  They started failing on Feb 23-24.
<img width="1229" alt="Screenshot 2023-02-27 at 3 49 59 PM" src="https://user-images.githubusercontent.com/5459654/221715486-f1c4f3cd-256f-4a49-9903-b71969d262da.png">

It's possible this is due to the latest change to run on Anyscale Jobs or the v2 stack.  Should we just revert that change for these three tests for now?

**Comment by architkulkarni** [comment]
This might be a distraction, but the timing seems to match up with this commit: https://github.com/ray-project/ray/pull/32784

**Comment by shomilj** [comment]
Looking at this now.

**Comment by architkulkarni** [comment]
Thanks -- the relevant context from the Ray side is that when a local directory path (e.g. `/code/my_dir`) is specified as `working_dir`, Ray zips it and uploads it to the GCS, and *rewrites* the `runtime_env`'s `working_dir` field to `gcs://ray_pkg_<hash>`, which is the URI for the package in the GCS.

There seems to be some interaction here where when a Ray Job is submitted from inside the entrypoint script of an Anyscale job, some Anyscale `checkpoint_job` code (not clear if this is supposed to be called) doesn't know how to interpret the `gcs://` URI in the `working_dir` field.

**Comment by shomilj** [comment]
Useful context from the Anyscale side: the codepath that's being executed is the codepath that enables you to clone an Anyscale Job into a Workspace.

The thing that changed when this test started failing was that we turned on this feature in staging -- it seems like this feature _specifically_ does not work super well when you call `ray.init("auto")` inside an Anyscale Job driver -- because then the hook executes again and finds a `gcs://` url and panics.

Code snippet from this test that I think is suspicious:
```
import time

import ray

# At this point, it looks like the Anyscale hook is executing, and failing because the 
# `working_dir` has been overwritten.
ray.init("auto")

@ray.remote(num_cpus=1)
class A:
    def f(self):
        return 1

actors = [A.remote() for _ in range(85)]

# Keep calling actor methods which will generate lots of metrics.
while True:
    time.sleep(0.1)
    ray.get([actor.f.remote() for actor in actors])
```

I think we never ran into this on the V1 stack since this test probably ran as a Ray client or SDK command test. So reverting to the V1 stack would unblock this for now.

So TLDR: is it necessary to run `ray.init("auto")` in the driver? I do not have enough context here.

If it is, we can revert to the V1 stack / SDK command execution mode while we figure out what the right fix is for this on the product or OSS side.

If it is not, then remove it and the test will probably start working.

**Comment by shomilj** [comment]
After discussing more - this might be a product regression. We're fixing it. Should be good to go w/o any changes in the release test by mid-day / EOD tomorrow.
